web/viewer-geckoview.html

```diff
@@ -0,0 +1,106 @@
+<!DOCTYPE html>
+<!--
+Copyright 2012 Mozilla Foundation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Adobe CMap resources are covered by their own copyright but the same license:
+
+    Copyright 1990-2015 Adobe Systems Incorporated.
+
+See https://github.com/adobe-type-tools/cmap-resources
+-->
+<html dir="ltr" mozdisallowselectionprint>
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
+<!--#if GENERIC || CHROME-->
+    <meta name="google" content="notranslate">
+<!--#endif-->
+    <title>PDF.js viewer</title>
+
+<!--#if MOZCENTRAL-->
+<!--#include viewer-snippet-firefox-extension.html-->
+<!--#endif-->
+
+<!--#if MOZCENTRAL-->
+<link rel="stylesheet" href="viewer.css">
+<!--#else-->
+<link rel="stylesheet" href="viewer-geckoview.css">
```

Comment:
Don't you want to include this in the MOZCENTRAL build as well?

---

Is the GeckoView-viewer ever going to be used for anything except Firefox, and as a GENERIC viewer during development?
Because if not, there's probably a number of pre-processor blocks that can now be simplified or even removed; possible a bit too much copy-and-paste from the regular viewer?

Reply:
> Don't you want to include this in the MOZCENTRAL build as well?

In m-c, we use `viewer.html` at several places, for example:
https://searchfox.org/mozilla-central/rev/17aeb39742eba71e0936ae44a51a54197100166d/toolkit/content/contentAreaUtils.js#115

so I'll map `viewer-geckoview.html` to `viewer.html` in the `jar.nm` file:
https://searchfox.org/mozilla-central/source/toolkit/components/pdfjs/jar.mn

and in order to have something consistent, I'll map the css and the js geckoview files either to keep `viewer.css` and `viewer.js`.